### DISCUSSION TITLE
save_report() still need Internet

### DISCUSSION THREAD

**Discussion by csu007** [datapane/datapane#368] [Help] [answered]
### Summary

save_report()
Reports can be saved / exported as a standalone file which can be shared without a server.

when i try to open  the a standalone file by chrome(111.0.5563.65)  without network, nothing show
i think the standalone file  still need something from https://datapane-cdn.com/.....

**Comment by mands** [comment, marked as answer]
Hi @csu007 ,

Yes, this is by design, we store all JS and CSS in a CDN to reduce the size of the local file - otherwise each report will have ~10MB of assets included within them that is duplicated in each saved report.

The wording in the docs is a little bit ambiguous and we'll clear that up, we mean it doesn't require a dedicated web server, i.e. you can open the local html file and it will work, not that it doesn't require any internet access.

**Reply by csu007** [reply]
thank you very much.  I can understand, a fully self-contained HTML file's size is really large.

**Reply by ya1sec** [reply]
Where is the data stored? Does saving a report to a local HTML file send the data to Datapane cloud?

**Reply by Buroni** [reply]
Hey @ya1sec any data used in the report is stored locally in the report HTML file. The resources used to display the data (e.g. plotting libraries) are loaded from our CDN.
